syntax = "proto3";

package zkp_auth;

message RegisterRequest {
  string user = 1;
  string y1 = 2;
  string y2 = 3;
}

message RegisterResponse {
  int32 status = 1;
}

message AuthenticationChallengeRequest {
  string user = 1;
  string r1 = 2;
  string r2 = 3;
}

message AuthenticationChallengeResponse {
  string auth_id = 1;
  int64 c = 2;
}

message AuthenticationAnswerRequest {
  string auth_id = 1;
  string s = 2;
}

message AuthenticationAnswerResponse {
  string session_id = 1;
}

service Auth {
  rpc Register(RegisterRequest) returns (RegisterResponse) {}
  rpc CreateAuthenticationChallenge(AuthenticationChallengeRequest) returns (AuthenticationChallengeResponse) {}
  rpc VerifyAuthentication(AuthenticationAnswerRequest) returns (AuthenticationAnswerResponse) {}
}
